File: packages/navigation/android/src/main/java/expo/modules/gaodemap/map/overlays/HeatMapViewModule.kt
package expo.modules.gaodemap.map.overlays

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * HeatMap 视图 Module
 */
class HeatMapViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("NaviHeatMapView")

    View(HeatMapView::class) {
      Prop<List<Map<String, Any>>>("data") { view: HeatMapView, data ->
        view.setData(data)
      }
      
      Prop<Int>("radius") { view: HeatMapView, radius ->
        view.setRadius(radius)
      }
      
      Prop<Double>("opacity") { view: HeatMapView, opacity ->
        view.setOpacity(opacity)
      }
    }
  }
}

File: packages/navigation/ios/map/overlays/HeatMapViewModule.swift
import ExpoModulesCore

public class NaviHeatMapViewModule: Module {
    public func definition() -> ModuleDefinition {
        Name("NaviHeatMapView")
        
        View(NaviHeatMapView.self) {
            Prop("data") { (view: NaviHeatMapView, data: [[String: Any]]) in
                view.setData(data)
            }
            
            Prop("radius") { (view: NaviHeatMapView, radius: Int) in
                view.setRadius(radius)
            }
            
            Prop("opacity") { (view: NaviHeatMapView, opacity: Double) in
                view.setOpacity(opacity)
            }
        }
    }
}